app/api/discount_codes.py

```diff
@@ -20,18 +23,23 @@ class Meta:
         self_view_kwargs = {'id': '<id>'}
         inflect = dasherize
 
+    @validates_schema
+    def validate_quantity(self, data):
+        if data['max_quantity'] < data['min_quantity']:
+            raise UnprocessableEntity({'pointer': 'max_quantity'}, "max_quantity should be greater than min_quantity")
+
     id = fields.Integer()
     code = fields.Str()
     discount_url = fields.Url()
     value = fields.Float()
-    type = fields.Str()
+    type = fields.Str(validate=validate.OneOf(choices=["amount", "percent"]))
     is_active = fields.Boolean()
-    tickets_number = fields.Integer()
-    min_quantity = fields.Integer()
-    max_quantity = fields.Integer()
+    tickets_number = fields.Integer(validate=lambda n: n >= 0)
+    min_quantity = fields.Integer(validate=lambda n: n >= 0)
+    max_quantity = fields.Integer(validate=lambda n: n >= 0)
```

Comment:
Is it possible to check and ensure this is higher (or) equal to the `min_quantity` and at the same time less than or equal to `tickets_number` ?